src/NServiceBus.Serverless/ServerlessEndpoint.cs

```diff
@@ -39,10 +38,11 @@ public async Task Process(MessageContext message, TExecutionContext executionCon
         /// <summary>
         /// Lets the NServiceBus pipeline process this failed message.
         /// </summary>
-        public async Task ProcessFailedMessage(MessageContext messageContext, ExceptionDispatchInfo exceptionInfo, int immediateProcessingAttempts, TExecutionContext executionContext)
+        public async Task<ErrorHandleResult> ProcessFailedMessage(MessageContext messageContext, Exception exception, int immediateProcessingAttempts, TExecutionContext executionContext)
```

Comment:
I'm not opposed to this change. One problem @timbussmann and I saw is that when we expose it this way we have essentially 1:1 exposed the pump API and then the logical next question would be why not both packages simply implement IPushMessages

Reply:
The same killer argument could have been used by me to oppose to the changes done on the merge so what's your point?

My point was that an abstraction has to serve a purpose. A higher level abstraction introduces higher level concepts. If those are not present then a valid question is what the purpose of the abstraction is. Because when we even start returning the enum we essentially have exposed both pipeline delegates one to one. And then the serverless package could also be reduced to be an abstract transport kit that wraps transports with manual pumps. Of course you can always say well is exposing an enum really the tipping point? Maybe it is, maybe not. Important is the explicit decision and that's what I'm trying to explore

Reply:
Like I said before in a call with you. As long as we have this split into serverless package vs downstreams that depend on it these questions are important and necessary because we are effectively developing a library. The focus changes if we are implementing the downstream changes only without the dependency.

Reply:
So I'm still curious to understand the reasons that made you change this PRs introduced behavior from throwing when retry required to return the result so that I can carry on that context if someone asks me